### DISCUSSION TITLE
[firebase_dynamic_links] Add FirebaseDynamicLinksCustomDomains information to iOS instructions

### DISCUSSION THREAD

**Discussion by aaronblondeau** [firebase/flutterfire#5578] [Feature request]
**Is your feature request related to a problem? Please describe.**
I lost an enormous amount of time trying to get dynamic links working on iOS - they would open the app, but getInitialLink always returned null.  I had setup firebase with a custom domain, and the documentation did not include information about adding FirebaseDynamicLinksCustomDomains in Info.plist.  I finally resolved the issue once I discovered instructions about this key here : https://firebase.google.com/docs/dynamic-links/ios/receive 

**Describe the solution you'd like**
Add a note about custom domain usage and FirebaseDynamicLinksCustomDomains in the iOS portion of "Handle Received Dynamic Links" to packages/firebase_dynamic_links/README.md

**Comment by LucaIaconelli** [comment]
Thanks I have been trying for days to understand why my dynamic links didn't work on iOS. It would be very helpful to add these instructions.

**Comment by jcsena** [comment]
@aaronblondeau  I have the same problem, the dynamic links open the app, but the getInitialLink always returned null. How did you solve them? Thanks..

**Comment by lukepighetti** [comment]
It looks like there was a breaking change. There used to be a method called `retreiveDynamicLink()` that would give you the most recent dynamic link. This appears to have been replaced with `getDynamicLink(Uri url)` which has no documentation and requires a Uri for some unspecified reason. It appears to not be possible to retrieve the most recent dynamic link. Additionally, the `onLink(...)` handler doesn't return a subscription so you can only use that at the top level of the app and only once as far as I know. If anyone has any info on these changes and what our options are now I would love to hear about them.

**Comment by purnomoeko** [comment]
any updates on this issue? ios dynamic link never works on my end, it always returns null when trying getDynamicLink(Uri uri)

**Comment by jcsena** [comment]
My workaround for that issue. it's work for me.

``` dart
  final PendingDynamicLinkData? data = await FirebaseDynamicLinks.instance.getInitialLink();
    //TODO: Workround for issue IOS dynamic links
  if (Platform.isIOS) {
    for (int i = 0; i < 8; i++) {
      await Future.delayed(Duration(milliseconds: 300));
      final PendingDynamicLinkData? _data =
          await FirebaseDynamicLinks.instance.getInitialLink();
      final _deepLink = _data?.link;
      if (_deepLink != null) {
        //catched dynamic link
        break;
      }
    }
  }

```
